.github/workflows/ci.yaml

```diff
@@ -255,6 +273,37 @@ jobs:
         with:
           name: test-reports-${{ matrix.abi }}-${{ env.TIMESTAMP }}
           path: android/app/build/outputs/integration_test_reports
+
+  rerun-workflow-job:
+    name: "Integration test - Rerun ${{ github.run_id }}"
+    runs-on: ubuntu-latest
+    needs: integration-test
+    if: failure() && ( needs.attempt_job.outputs.nextAttempt <= needs.attempt_job.outputs.maxAttempts )
+    env:
+      attempt: ${{ needs.attempt_job.outputs.nextAttempt }}
+    steps:
+      - name: "Trigger Attempt ${{ needs.attempt_job.outputs.nextAttempt }}"
+        shell: pwsh
+        env:
+          # GITHUB_TOKEN with actions: write will NOT work
+          # PAT Token must have workflow scope
+          GH_CLI_TOKEN: ${{ secrets.CI_RUNNER }}
+        run: |
+          echo "Running gh auth login..."
+          $env:GH_CLI_TOKEN | gh auth login --with-token
+          if(!$?) { exit 1 }
+          echo "Triggering rerun attempt $env:nextAttempt"
+          echo "
+          gh workflow run rerun-ci.yaml
```

Comment:
so do we need to seperate, for example, integration tests into its own workflow to get this to work?